pyravendb/store/document_store.py

```diff
@@ -98,10 +98,11 @@ def get_request_executor(self, db_name=None):
         db_name = db_name if db_name is not None else self.database
         if db_name is None:
             raise ValueError("database name cannot be None")
-
-        with self.lock:
-            self._request_executors.setdefault(db_name, RequestsExecutor.create(self.urls, db_name, self._certificate,
-                                                                                self.conventions))
+        if self._request_executors.get(db_name) is None:
```

Comment:
We are mutating and reading the dic outside the lock.
This _looks_ wrong.

Can you ensure that this won't cause race conditions? 
Can you check inside the lock as well?

Reply:
Thanks for the heads-up!!, will add a new test as well.